src/main/java/fr/esgi/al5/tayarim/services/LogementService.java

```diff
@@ -0,0 +1,344 @@
+      throw new LogementNotFoundException();
+    }
+
+    return LogementMapper.entityToDto(optionalLogement.get());
+  }
+
+  /**
+   * Met à jour les informations d'un logement existant.
+   *
+   * @param id                L'identifiant du logement à mettre à jour.
+   * @param logementUpdateDto Les nouvelles informations du logement.
+   * @return Le DTO du logement mis à jour.
+   * @throws ProprietaireInvalidUpdateBody si les données de mise à jour sont invalides.
+   */
+  @Transactional
+  public LogementDto updateLogement(@NonNull Long id,
+      @NonNull LogementUpdateDto logementUpdateDto) {
+
+    if ((logementUpdateDto.getTitre() == null || logementUpdateDto.getTitre().isBlank())
+        && (logementUpdateDto.getNombresDeChambres() == null
+        || logementUpdateDto.getNombresDeChambres() == 0)
+        && (logementUpdateDto.getNombresDeLits() == null
+        || logementUpdateDto.getNombresDeLits() == 0)
+        && (logementUpdateDto.getNombresSallesDeBains() == null
+        || logementUpdateDto.getNombresSallesDeBains() == 0)
+        && (logementUpdateDto.getCapaciteMaxPersonne() == null
+        || logementUpdateDto.getCapaciteMaxPersonne() == 0)
+        && (logementUpdateDto.getNombresNuitsMin() == null
+        || logementUpdateDto.getNombresNuitsMin() == 0)
+        && (logementUpdateDto.getDescription() == null || logementUpdateDto.getDescription()
+        .isBlank())
+        && (logementUpdateDto.getPrixParNuit() == null || logementUpdateDto.getPrixParNuit() == 0)
+        && (logementUpdateDto.getDefaultCheckIn() == null || logementUpdateDto.getDefaultCheckIn()
+        .isBlank())
+        && (logementUpdateDto.getDefaultCheckOut() == null || logementUpdateDto.getDefaultCheckOut()
+        .isBlank())
+        && (logementUpdateDto.getIntervalReservation() == null
+        || logementUpdateDto.getIntervalReservation() == 0)
+        && (logementUpdateDto.getVille() == null || logementUpdateDto.getVille().isBlank())
+        && (logementUpdateDto.getAdresse() == null || logementUpdateDto.getAdresse().isBlank())
+        && (logementUpdateDto.getCodePostal() == null || logementUpdateDto.getCodePostal()
+        .isBlank())
+        && (logementUpdateDto.getPays() == null || logementUpdateDto.getPays().isBlank())
+        && (logementUpdateDto.getEtage() == null || logementUpdateDto.getEtage().isBlank())
+        && (logementUpdateDto.getNumeroDePorte() == null || logementUpdateDto.getNumeroDePorte()
+        .isBlank())
+        && (logementUpdateDto.getIdTypeLogement() == null
+        || logementUpdateDto.getIdTypeLogement() == 0)) {
+      throw new LogementInvalidUpdateBody();
+    }
+
+    Optional<Logement> optionalLogement = logementRepository.findById(id);
+
+    if (optionalLogement.isEmpty()) {
+      throw new LogementNotFoundException();
+    }
+
+    Logement logement = optionalLogement.get();
+    double latitude = logement.getLatitude();
+    double longitude = logement.getLongitude();
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'longitude' declaration and its first usage is 18, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).